Pod/Classes/WPAssetViewController.m

```diff
@@ -61,6 +61,75 @@ - (void)viewDidLoad
     [self showAsset];
 }
 
++ (NSDateFormatter *)sharedDateFormater {
+    static NSDateFormatter * _sharedDateFormatter;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        _sharedDateFormatter = [[NSDateFormatter alloc] init];
+        _sharedDateFormatter.dateStyle = NSDateFormatterLongStyle;
+        _sharedDateFormatter.timeStyle = NSDateFormatterNoStyle;
+    });
+    return _sharedDateFormatter;
+}
+
++ (NSDateFormatter *)sharedTimeFormater {
+    static NSDateFormatter * _sharedTimeFormatter;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        _sharedTimeFormatter = [[NSDateFormatter alloc] init];
+        _sharedTimeFormatter.dateStyle = NSDateFormatterNoStyle;
+        _sharedTimeFormatter.timeStyle = NSDateFormatterShortStyle;
+    });
+    return _sharedTimeFormatter;
+}
+
++ (NSDateFormatter *)sharedDateWeekFormater {
+    static NSDateFormatter * _sharedDateWeekFormatter;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        _sharedDateWeekFormatter = [[NSDateFormatter alloc] init];
+        _sharedDateWeekFormatter.dateFormat = [NSDateFormatter dateFormatFromTemplate:@"cccc" options:0 locale:nil];
+    });
+    return _sharedDateWeekFormatter;
+}
+
+
+- (void)viewDidAppear:(BOOL)animated {
+    [super viewDidAppear:animated];
+    [self updateNavigationTitle];
+}
+
+- (void)updateNavigationTitle {
+    if (self.asset.date == nil || self.navigationController == nil) {
+        return;
+    }
+    UILabel *titleLabel = [[UILabel alloc] init];
+    titleLabel.textColor = self.navigationController.navigationBar.tintColor;
+
+    NSDate *assetDate = self.asset.date;
+    NSDate *now = [NSDate date];
+    NSString *dateString = [[[self class] sharedDateFormater] stringFromDate:assetDate];
+    NSString *timeString = [[[self class] sharedTimeFormater] stringFromDate:assetDate];
+
+    NSCalendar *gregorianCalendar = [NSCalendar calendarWithIdentifier:NSCalendarIdentifierGregorian];
```

Comment:
In fact, we probably shouldn't hardcode `7` for the number of days in the week either... something like this maybe?

```
NSDate *oneWeekAgo = [[NSCalendar currentCalendar] dateByAddingUnit:NSCalendarUnitWeekOfYear value:-1 toDate:now options:0];
if ([assetDate compare:oneWeekAgo] == NSOrderedDescending) {
    dateString = [[[[self class] sharedDateWeekFormater] stringFromDate:assetDate] capitalizedStringWithLocale:nil];
}
```